File: js/src/sorting/Sorter.js
class Sorter {
    constructor(arr = []) {
        this.arr = arr;
    }

    less(i, j) {
        return i < j;
    }

    exch(i, j) {
        let temp = this.arr[i];
        this.arr[i] = this.arr[j];
        this.arr[j] = temp;
    }

    show() {
        for (let i = 0; i < this.arr.length; i++) {
            console.log(this.arr[i] + ' ');
        }
    }
}

module.exports = Sorter;

File: java/src/com/test/sorting/Sorter.java
package com.test.sorting;


public abstract class Sorter
{
    protected int[] arr;

    Sorter(int[] arr) {
        this.arr = arr;
    }

    abstract  public void sort();

    protected boolean less(Integer a, Integer b)
    {
        return a < b;
    }

    protected void exch(int i, int j)
    {
        Integer temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public void show()
    {
        for (Integer item: arr) {
            System.out.print(item + " ");
        }

        System.out.println();
    }
}
